velox/docs/functions/spark/regexp.rst

```diff
@@ -57,8 +59,10 @@ See https://github.com/google/re2/wiki/Syntax for more information.
 .. spark:function:: regexp_replace(string, pattern, overwrite, position) -> varchar
     :noindex:
 
-    Replaces all substrings in ``string`` that match the regular expression ``pattern`` with the string ``overwrite`` starting from the specified ``position``. If the ``position`` is less than one, the function returns an error. If ``position`` is greater than the length of ``string``, the function returns the original ``string`` without any modifications.
-    There is a limit to the number of unique regexes to be compiled per function call, which is 20.
+    Replaces all substrings in ``string`` that match the regular expression ``pattern`` with the string ``overwrite`` starting from the specified ``position``.  If no match is found, the original string is returned as is. If the ``position`` is less than one, the function throws an exception. If ``position`` is greater than the length of ``string``, the function returns the original ``string`` without any modifications.
+    There is a limit to the number of unique regexes to be compiled per function call, which is 20. If this limit is exceeded the function will throw an exception.
+    regexp_replace will throw an exception if any provided arguments are null, if ``string`` contains an invalid UTF-8 character, if ``position`` is less than 1, or if ``pattern`` does not conform to RE2 syntax: https://github.com/google/re2/wiki/Syntax.
```

Comment:
> regexp_replace will throw an exception if any provided arguments are null

This is surprising behavior. Does it match Spark?

Reply:
Just confirmed. This function has default null behavior in Spark, i.e., returns null for any null input. I note the null behavior is already aligned with this Spark. @codyschierbeck, please correct the doc. Thanks!

Reply:
Apologies, this was doc mistake when trying to remove all the references to returning null that should have been throwing an exception. Adjusting.